bpf/lib/nat.h

```diff
@@ -1139,6 +1305,86 @@ snat_v6_rev_nat_handle_mapping(struct __ctx_buff *ctx,
 		       NAT_PUNT_TO_STACK : DROP_NAT_NO_MAPPING;
 }
 
+static __always_inline int snat_v6_icmp_rewrite_embedded(struct __ctx_buff *ctx,
+							 struct ipv6_ct_tuple *tuple,
+							 struct ipv6_nat_entry *state,
+							 __u32 l4_off)
+{
+	struct csum_offset csum = {};
+	int hdrlen;
+	__u32 inner_l4_off;
+
+	hdrlen = ipv6_hdrlen(ctx, &tuple->nexthdr);
```

Comment:
This also needs to be `ipv6_hdrlen_offset()`.

But I don't understand why it's needed in the first place - the caller just extracted the inner CT tuple, so it *must* know the `inner_l3_off` and `inner_l4_off`. Why not simply pass those through?
Same applies for the corresponding IPv4 code.

Reply:
++